drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/SubHeaderBuilder.java

```diff
@@ -0,0 +1,174 @@
+                makeAction(baseColumn);
+            } else if (baseColumn instanceof RowNumberCol52) {
+                // Ignore and do not add to count
+                continue;
+            } else if (baseColumn instanceof DescriptionCol52) {
+                // This is actually a column, but header is not written down in XLS
+            } else {
+                throw new IllegalArgumentException("TODO REMOTE THIS");
+            }
+            columnIndex++;
+        }
+    }
+
+    private void makeAction(final BaseColumn baseColumn) {
+
+        if (baseColumn instanceof ActionSetFieldCol52) {
+            makeSetField((ActionSetFieldCol52) baseColumn);
+        } else if (baseColumn instanceof ActionInsertFactCol52) {
+            makeInsert((ActionInsertFactCol52) baseColumn);
+        } else if (baseColumn instanceof ActionRetractFactCol52) {
+            makeRetract((ActionRetractFactCol52) baseColumn);
+        }
+    }
+
+    private void makeHeaderAndTitle(final String action,
+                                    final String header) {
+        headerRow.createCell(columnIndex)
+                .setCellValue(action);
+
+        headerTitleRow.createCell(columnIndex)
+                .setCellValue(header);
+    }
+
+    private void makeRetract(final ActionRetractFactCol52 column) {
+
+        makeHeaderAndTitle(ACTION,
+                           column.getHeader());
+        fieldRow.createCell(columnIndex).setCellValue("retract( $param );");
+    }
+
+    private void makeInsert(final ActionInsertFactCol52 column) {
+
+        if (!addedInserts.contains(column.getBoundName())) {
+
+            makeHeaderAndTitle(ACTION,
+                               "");
+
+            fieldRow.createCell(columnIndex).setCellValue(column.getFactType() + " " + column.getBoundName() + " = new " + column.getFactType() + "(); insert( " + column.getBoundName() + " );");
+
+            addedInserts.add(column.getBoundName());
+            columnIndex++;
+        }
+
+        makeHeaderAndTitle(ACTION,
+                           column.getHeader());
+
+        fieldRow.createCell(columnIndex).setCellValue(makeSetMethod(column.getBoundName(), column.getFactField()));
+    }
+
+    private void makeSetField(final ActionSetFieldCol52 column) {
+
+        makeHeaderAndTitle(ACTION,
+                           column.getHeader());
+        fieldRow.createCell(columnIndex).setCellValue(makeSetMethod(column.getBoundName(), column.getFactField()));
+    }
+
+    private void makeCondition(final ConditionCol52 column) {
+
+        makeHeaderAndTitle(CONDITION,
+                           column.getHeader());
+
+        fieldRow.createCell(columnIndex).setCellValue(String.format("%s %s $param", column.getFactField(), column.getOperator()));
+    }
+
+    private void makeAttribute(final AttributeCol52 column) throws ConversionException {
+        if (Objects.equals("negate", column.getAttribute().toLowerCase())) {
```

Comment:
### Manual Review Note 1
We export other boolean attributes (for example **enabled**) as 0/1, is it expected or we want to throw an exception for all boolean attributes?

Reply:
Other should work. Need to check what is going on.

Reply:
I'm getting "TRUE" when using enabled.